File: CatAndMouse.java
// https://www.hackerrank.com/challenges/cats-and-a-mouse/problem?isFullScreen=true

class CatAndMouse{
    public static String catAndMouse(int x, int y, int z){
        int catAdiff = Math.abs(z - x);
        int catBdiff = Math.abs(z - y);
        
        if(catAdiff < catBdiff){
            return "Cat A";
        }
        else if(catBdiff < catAdiff){
            return "Cat B";
        }
        else{
            return "Mouse C";
        }
    }

    public static void main(String[] args) {
        int x = 1; // Cat A position
        int y = 2; // Cat B position
        int z = 3; // Mouse C position

        String result = catAndMouse(x, y, z);
        System.out.println(result); // Output: Cat B
    }
}

File: Python/CatAndMouse.py
# https://www.hackerrank.com/challenges/cats-and-a-mouse/problem?isFullScreen=true

class solution:
    def catAndMouse(x, y, z):
        catAdiff = abs(z - x)
        catBdiff = abs(z - y)
        
        if(catAdiff < catBdiff):
            return "Cat A"
        elif(catBdiff < catAdiff):
            return "Cat B"
        else:
            return "Mouse C"
        
    if __name__ == "__main__":
        x = 1
        y = 2
        z = 3
        print(catAndMouse(x, y, z))
